Sentinel/Classes/Core/PreferencesInfo/PreferenceItem.swift

```diff
@@ -0,0 +1,194 @@
+        case integer(min: Int?, max: Int?)
+        /// Double input with minimum and maximum values
+        case double(min: Double?, max: Double?)
+        /// Enum input, as a RawRepresentable with either String or Int as type.
+        case enumeration(allCases: [any RawRepresentable])
+    }
+
+    /// Name of the item
+    let name: String
+
+    /// More information. E.g. instructions, format etc.
+    let info: String?
+
+    /// The type of this preference
+    let type: InputType
+
+    /// Value, saved as string.
+    var value: String? {
+        get { userDefaults.string(forKey: userDefaultsKey) }
+        set { userDefaults.set(newValue, forKey: userDefaultsKey) }
+    }
+
+    // MARK: - Private Properties
+
+    private let userDefaults: UserDefaults
+    private let userDefaultsKey: String
+    private let onDidSet: ((String?) -> Void)?
+
+    // MARK: - Methods
+
+    /// Initializes and returns a new Preference.
+    ///
+    /// - Parameter name: The name of the preference
+    /// - Parameter info: Optional additional info like instructions, format etc.
+    /// - Parameter type: The type of this preference.
+    /// - Parameter userDefaultsKey: The key under which this preference is saved in UserDefaults
+    /// - Parameter userDefaults: Optionally define a different suite of UserDefaults (standard by default).
+    /// - Parameter onDidSet: Optional execution block once the value is changed.
+    init(
+        name: String,
+        info: String? = "",
+        type: InputType,
+        userDefaultsKey: String?,
+        userDefaults: UserDefaults = .standard,
+        onDidSet: ((String?) -> Void)?
+    ) {
+        self.name = name
+        self.info = info
+        self.type = type
+        self.userDefaultsKey = userDefaultsKey ?? ""
+        self.userDefaults = userDefaults
+        self.onDidSet = onDidSet
+    }
+
+    // MARK: - Public Initializers
+
+    /// Please see `PreferenceItem+Init` for all the public initiailzers.
+
+    // MARK: - Internal Methods
+
+    func didChangeInputValue(newValue: String?) throws {
+        try store(newValue: newValue)
+    }
+
+    func store(newValue: String?) throws {
+        guard var newValue else {
+            userDefaults.set(nil, forKey: userDefaultsKey)
+            return
+        }
+        switch type {
+        case .bool:
+            value = String(newValue.boolValue)
+        case .text:
+            value = newValue
+        case .integer(let minimum, let maximum):
+            guard let number = Int(newValue) else {
+                throw InputError.formatError
+            }
+            if let minimum, let maximum, number > maximum || number < minimum {
+                throw InputError.outOfRange
+            }
+            value = String(number)
+        case .double(let minimum, let maximum):
+            newValue = newValue.replacingOccurrences(of: ",", with: ".") // Swift only recognizes dot as decimal separator.
```

Comment:
Are you sure about that? Can you test this with different regional settings?

Reply:
Yup, the Double initialiser only accepts dot as the decimal separator, irregardless of region.
So many regions use comma so I added this as a tiny improvement, not planning on implementing number formatter here really.